### PR TITLE
Add unit tests for Core IdentifiableObject classes

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add unit tests for Core IdentifiableObject classes
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis should be green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12304)

### CODE REVIEW

File: tests/Unit/Core/Form/IdentifiableObject/Handler/FormHandlerTest.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            ->method('dispatchWithParameters')
+            ->with(
+                $this->equalTo('actionAHookSave'),
+                $this->equalTo([
+                    'errors' => ['err' => 'or'],
+                    'form_data' => ['x' => 'y'],
+                ])
+            );
+
+        $this->handler->save(['x' => 'y']);
```
Comment (Quetzacoalt91) [inline tests/Unit/Core/Form/IdentifiableObject/Handler/FormHandlerTest.php:139]: Didn't you forget an `assert` statement? Or you only want to check the call to `dispatchWithParameters()`?

Comment (matks) [reply tests/Unit/Core/Form/IdentifiableObject/Handler/FormHandlerTest.php:139]: The return of `$this->handler->save(...)` is an array `$errors` directly taken from the DataProvider which is a mock here. So I thought it was not relevant to check, but I can add it.

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Form/FormHandler.php#L123
```
public function save(array $data)
    {
        $errors = $this->formDataProvider->setData($data);
        $this->hookDispatcher->dispatchWithParameters(
            "action{$this->hookName}Save",
            [
                'errors' => &$errors,
                'form_data' => &$data,
            ]
        );
        return $errors;
    }
```

Comment (Quetzacoalt91) [reply tests/Unit/Core/Form/IdentifiableObject/Handler/FormHandlerTest.php:139]: Nope, that's fine by me.

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by Quetzacoalt91** [comment]
No need of QA, merging.

**Pull request merged**